elbepack/initvmaction.py

```diff
@@ -269,7 +392,33 @@ class AttachAction(InitVMAction):
     def __init__(self, node):
         InitVMAction.__init__(self, node)
 
-    def execute(self, _initvmdir, _opt, _args):
+    def _attach_qemu_vm(self, initvmdir):
+        """
+        Attach to QEMU initvm.
+
+        This method is using socat to connect to the unix-socket of the 
+        serial console of the initvm.
+        """
+        initvmdir = check_initvm_dir(initvmdir)
+
+        # Test if socat command is available.
+        try:
+            subprocess.run('which socat > /dev/null 2>&1', shell=True, check=True)
+        except Exception:
+            print('The command "socat" is required.\nPlease install socat: "sudo apt install socat"')
+            sys.exit(208)
+        
+        # Connect to socket file, if it exists.
+        if os.path.exists(os.path.join(initvmdir, 'vm-serial-socket')):
+            subprocess.run(f'cd {initvmdir} && socat stdin,raw,echo=0,escape=0x11 unix-connect:vm-serial-socket',
```

Comment:
Why `check=False` and `shell=True`? Also use `.run(cwd=initvmdir)`

Reply:
AFAIK it's only possible to end this "socat session" with sending a kill to the process or stopping the VM sind all keys are forwarded to the VM  (which is the same for the virsh console).
In case of stopping the VM, the exit code of socat is 0, which is fine.
in case of sending the kill signal, the exit code is != 0, which will cause a CalledProcessError if check is true.

Reply:
Shell is indeed not needed, and I will remove it.